File: 6-prefix-sum/2-cumulative-sums/main.cpp
// Cumulative Sums - C++ Example

#include <iostream>
#include <vector>

std::vector<int> cumulativeCountPositive(const std::vector<int>& arr) {
    std::vector<int> cumCount(arr.size());
    cumCount[0] = arr > 0 ? 1 : 0;
    for (size_t i = 1; i < arr.size(); i++) {
        cumCount[i] = cumCount[i - 1] + (arr[i] > 0 ? 1 : 0);
    }
    return cumCount;
}

int main() {
    std::vector<int> arr = {-1, 4, 2, -3, 5};
    std::vector<int> countPos = cumulativeCountPositive(arr);

    std::cout << "Cumulative count of positives: ";
    for (int val : countPos) std::cout << val << " ";
    std::cout << std::endl;
    return 0;
}

File: 6-prefix-sum/2-cumulative-sums/main.py
# Cumulative Sums - Python Example

def cumulative_count_positive(arr):
    cum_count = [0] * len(arr)
    cum_count = 1 if arr > 0 else 0
    for i in range(1, len(arr)):
        cum_count[i] = cum_count[i - 1] + (1 if arr[i] > 0 else 0)
    return cum_count

arr = [-1, 4, 2, -3, 5]
count_pos = cumulative_count_positive(arr)
# print("Cumulative count of positives:", count_pos)
